### ISSUE TITLE
Can't run Fable in watch mode

### ISSUE THREAD

**Issue by teknikal-wizard** [alfonsogarciacaro/fable-py-sample#3]
I'm just trying out the Python Interactive sample and I get
```bash
PS F:\Desktop\fable-py-sample-python-interactive\fable-py-sample-python-interactive> dotnet fable watch --lang py
Fable: F# to Python compiler 4.0.0-theta-010 (status: beta)
Thanks to the contributor! @inchingforward
Stand with Ukraine! https://standwithukraine.com.ua/

Parsing program.fsx...
Project and references (2 source files) parsed in 905ms

Unhandled exception. System.Exception: No common base dir
   at Fable.Cli.Main.FileWatcherUtil.getCommonDir@210(FSharpList`1 restDirs, String dir) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Main.fs:line 218
   at Fable.Cli.Main.FileWatcherUtil.getCommonBaseDir(FSharpList`1 files) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Main.fs:line 220
   at Fable.Cli.Main.FsWatcher.Observe(FSharpList`1 filesToWatch) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Main.fs:line 258
   at Fable.Cli.Main.Watcher.Watch(ProjectCracked projCracked) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Main.fs:line 575
   at Fable.Cli.Main.compilationCycle@731.Invoke(Unit unitVar) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Main.fs:line 769
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvoke[T,TResult](AsyncActivation`1 ctxt, TResult result1, FSharpFunc`2 part2) in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 510
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction) in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 112
--- End of stack trace from previous location ---
   at Microsoft.FSharp.Control.AsyncResult`1.Commit() in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 454
   at Microsoft.FSharp.Control.AsyncPrimitives.QueueAsyncAndWaitForResultSynchronously[a](CancellationToken token, FSharpAsync`1 computation, FSharpOption`1 timeout) in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 1140
   at Microsoft.FSharp.Control.AsyncPrimitives.RunSynchronously[T](CancellationToken cancellationToken, FSharpAsync`1 computation, FSharpOption`1 timeout) in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 1167
   at Microsoft.FSharp.Control.FSharpAsync.RunSynchronously[T](FSharpAsync`1 computation, FSharpOption`1 timeout, FSharpOption`1 cancellationToken) in /home/dev/Projects/fsharp/src/FSharp.Core/async.fs:line 1511
   at Fable.Cli.Entry.startCompilation@292-16(CliArgs cliArgs, FSharpOption`1 watchDelay, Unit unitVar0) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Entry.fs:line 293
   at Fable.Cli.Entry.Run@214-2.Invoke(Unit _arg5) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Entry.fs:line 297
   at Fable.Cli.Entry.main@386.Invoke(Tuple`2 _arg2) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Entry.fs:line 392
   at Fable.Cli.Entry.main(String[] argv) in /Users/dbrattli/Developer/Github/Fable/src/Fable.Cli/Entry.fs:line 384
```

**Comment by teknikal-wizard** [comment]
Experimented a a bit further, and can minimally recreate 
- start with an empty directory
- create an empty `.fsx` script.
- create a dotnet tool manifest and run `dotnet tool install fable --local --version "4.0.0-theta-*"`
- start Fable using `dotnet fable watch --lang py`
- watch mode starts ok
- add `#r "nuget:Fable.Core, 4.0.0-theta-004"` to the fsx
- stop fable watch with Ctrl-C
- try to start it again with `dotnet fable watch --lang py`
- observe the above `no common base dir` error

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for investigating this @teknikal-wizard! I can reproduce. When running with `--verbose` option I realized the problem is the F# compiler introduces some generated files to resolve Nuget references in .fsx scripts, and this confuses Fable algorithm to get the common base directory to watch.

The generated files used to end with ".fsproj.fsx" but we've recently updated FSharp.Compiler.Service and now they seem to be in a `.packagemanagement` directory. I've added a rule to ignore these and it seems to work now 👍
